lib/flecks/queue.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class Flecks::Queue
+	def initialize
+		@queue = ::Queue.new
+		@count = 0
+	end
+
+	def push(id, task, content)
+		@count += 1
+
+		Async do
+			@queue.push([id, task.call, content])
+		end
+	end
```

Comment:
This takes a task as a plain old proc and makes it asynchronously run the proc and push it into the queue.